### ISSUE TITLE
Sync fail sliently when S3 region field is empty

### ISSUE THREAD

**Issue by jerrylususu** [laurent22/joplin#6034]
I noticed today that when I click on the sync icon on the left-bottom corner, it won't start the sync process. After open the devtool, the problems seems to be in the S3 sync setting. As I am using a S3-compatible service as my sync target, it doesn't really have a `region` property, so the `region` field in the settings page remains empty, which causes the `Region is missing` error. 

Surprisingly, I was not aware of that error, as there is no obvious signs in the UI showing that my sync failed, and the failure is completely silent beside the error messages that can only be found when devtool is opened. What is more, when click on the `check sync configuration` button, it just shows `Checking, please wait` and stalls here.

Setting a region to a non-empty string (such as `anything`) seems to fix the `Region is missing` error. However, it should be obvious when a sync failure happens, so the user can take appropriate actions to fix that.

## Environment

Joplin version:  2.6.10 (prod, win32), revision 98fba37
Platform: Windows
OS specifics: Windows 11 x64 Pro

## Steps to reproduce

1. Use S3 (Beta) as sync target, but leave the `Region` field empty.
2. Press the `check sync configuration` button, and see it stuck at `Checking, please wait`. Or
3. Go to main page, click on the `Sync` button, and nothing happens.

## Describe what you expected to happen
A warning message indicating sync has failed with diagnose details.

## Logfile

```
Uncaught (in promise) Error: Region is missing
    at Object.resolveRegionConfig (C:\Users\User\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\node_modules\@aws-sdk\config-resolver\dist-cjs\regionConfig\resolveRegionConfig.js:9)
    at new S3Client (C:\Users\User\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\node_modules\@aws-sdk\client-s3\dist-cjs\S3Client.js:20)
    at Function.newFileApi_ (C:\Users\User\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\SyncTargetAmazonS3.js:88)
    at Function.checkConfig (C:\Users\User\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\SyncTargetAmazonS3.js:101)
    at Object.shared.checkSyncConfig (C:\Users\User\AppData\Local\Programs\Joplin\resources\app.asar\node_modules\@joplin\lib\components\shared\config-shared.js:32)
    at ConfigScreenComponent.<anonymous> (C:\Users\User\AppData\Local\Programs\Joplin\resources\app.asar\gui\ConfigScreen\ConfigScreen.js:57)
    at Generator.next (<anonymous>)
    at C:\Users\User\AppData\Local\Programs\Joplin\resources\app.asar\gui\ConfigScreen\ConfigScreen.js:8
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\User\AppData\Local\Programs\Joplin\resources\app.asar\gui\ConfigScreen\ConfigScreen.js:4)
```

**Comment by k3d3** [comment]
I also see this issue. I'm using B2, and it fails if my region is blank. It works if I put "region" as my region, so I figure it works with any string.

**Comment by jerrylususu** [comment]
After reading the pinned issue regarding S3 sync (#5875), it seems that this bug is already fixed by PR #5923, so this should be resolved in next version of Joplin. I am closing this issue now.
